Aptoide/aptoide-client-v8
# [feature/ASV-1736] Load Wallet State

**What does this PR do?**

   This PR handles the initial load of the dialog dependencies, i.e. the app icon and the wallet app. It extracts the wallet app get to PromotionsManager. It also introduces minor layout changes that were needed to support both portrait and landscape modes.

Note: ASV-1734 and ASV-1735 needs to be merged first.

**Database changed?**

   No

**How should this be manually tested?**

  Send a relevant intent and check if the app icon is loaded correctly and if the layout is ok both in portrait and landscape mode. Also check if the wallet is loaded correctly (you can only see this by debugging).

   As a prerequisite, you need to send an intent relative to an app that you have installed (otherwise the icon cannot be retrieved and the dialog is automatically dismissed). Currently, the dialog should be dismissed (or not shown at all) in case of any error.

   For example, if you have Legacy of Discord installed, you can test with the following intent:
`adb shell am start -a "android.intent.action.VIEW" "market://details?id=com.appcoins.wallet\&utm_source=myappcoins\&app_source=com.gtarcade.lod"`

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1736](https://aptoide.atlassian.net/browse/ASV-1736)


**Code Review Checklist**
